galaxy.yml

```diff
@@ -9,6 +9,8 @@ authors:
 tags:
   - infrastructure
 repository: https://github.com/kubernetes-sigs/kubespray
+dependencies:
```

Comment:
I'm wondering what we should do about dependencies in general here, we have some check mostly on ansible core that is required to be 2.14.x currently. I'm wondering how we should handle these things in galaxy though. Do you have insight about that? Maybe we could also add ansible-core there?

Reply:
Hi Arthur, long time no see indeed :)
I don't think ansible-core should go there.
From [ansible docs on the subject](https://docs.ansible.com/ansible/latest/dev_guide/developing_collections_shared.html#listing-collection-dependencies), it looks like it's expected to treat ansible-core specially with the meta/runtime.yml file, and all other collections in the dependencies keys in galaxy.yml.

Note that they encourage collections authors to **not** depend on other collections, but I don't think kubespay should feel too concerned by that.

Reply:
Ok so I checked and I think ansible-core would be the same as the collection ansible.builtin. It seems to follow the same version (~ish). Could you add this to dependencies: `ansible.builtin: ">= 2.14.0 <2.15.0` (there's a bug in ansible-core 2.15 preventing to run kubespray currently).

Reply:
The way the above documentation is written, it looks to me that ansible-core *should* be treated specially and not put in `dependencies`.

> We recommend that collections work as standalone, independent units, depending only on ansible-core. However, if your collection must depend on features and functionality from another collection, list the other collection or collections under `dependencies `in your collection’s `galaxy.yml` file. Use the `meta/runtime.yml` file to set the ansible-core version that your collection depends on.

To me that reads like deps check for ansible-core specifically should go in `meta/runtime.yml`, as it is currently.

Reply:
Hmmm ok but then I'm wondering it is really useful to have a requirement on ansible.utils, shouldn't the minimum version of Ansible implicitly include that already somehow ?

Reply:
Apparently you can install ansible-core standalone, (but installing ansible would indeed have ansible.utils I think)

Reply:
Okok well then nvm both are fine as you want 👍

Reply:
And thanks for the valuable insights into this 🙏